### PR TITLE
Cart entity implementation with relations

### PR DESCRIPTION
**Pull request by annapawlas**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @OneToOne(mappedBy = "cart")
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @OneToOne(mappedBy = "cart")
+    private Order order;
+
+    @ManyToMany(mappedBy = "cartList")
+    private List<Product> listOfProducts  = new ArrayList<>();
```
Comment (JakubFir) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: We need relation to the linking table becouse we’re adding quantity column

**Review by JakubFir — CHANGES_REQUESTED** [review]

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 1 above)
Comment (annapawlas) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: I have created a CartProducts entity that represents the relationship between the Cart and Product entities. Please re-review.

File: src/main/java/com/kodilla/ecommercee/domain/CartProducts.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @JoinColumn(name = "cartId")
+    private Cart cartId;
+
+    @Column(name = "quantity")
+    private int quantity;
+
+    @OneToMany(
+            fetch = FetchType.LAZY,
+            mappedBy = "order")
+    private List<Order> orders;
```
Comment (JakubFir) [inline src/main/java/com/kodilla/ecommercee/domain/CartProducts.java]: oh am sorry, my bad, there is a bug in EDR diagram the cardinalities should be in reverse and it should be @ManyToOne

File: src/main/java/com/kodilla/ecommercee/domain/CartProducts.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    @Column(name = "cartProductsId", unique = true, nullable = false)
+    private Long cartProductsId;
+
+    @ManyToOne
+    @JoinColumn(name = "productId")
+    private Product product;
+
+    @ManyToOne
+    @JoinColumn(name = "cartId")
+    private Cart cartId;
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/domain/CartProducts.java]: I think field in entity should be "cart". "cartId" is the name of column in table

File: src/main/java/com/kodilla/ecommercee/domain/Order.java [hunk 4]
```diff
@@ -19,4 +20,15 @@ public class Order {
     @Enumerated(EnumType.STRING)
     @Column
     private OrderStatus orderStatus;
+
+    @OneToOne()
+    private Cart cart;
+
+    @ManyToOne
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/domain/Order.java]: In reverse, as @JakubFir wrote

File: src/main/java/com/kodilla/ecommercee/domain/Product.java [hunk 5]
```diff
@@ -30,4 +32,18 @@ public class Product {
     @JoinColumn(name = "groupId")
     private Group group;
     */
+
+    @OneToMany(
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/domain/Product.java:36]: In reverse

**Review by sew-k — APPROVED** [review]

**Review by WorstWorm — APPROVED** [review]
:thumbsup:

**Review by BartoszSzych98 — APPROVED** [review]

**Review by JakubFir — APPROVED** [review]

**Pull request merged**